src/session/session_api.c

```diff
@@ -1560,6 +1562,24 @@ __wt_session_range_truncate(
         WT_ERR(__wt_buf_set(session, orig_stop_key, stop_key.data, stop_key.size));
     }
 
+    /*
+     * Now that the truncate is setup and ready regardless of how the API was called, populate our
+     * truncate information cookie.
+     */
+    trunc_info->session = session;
+    trunc_info->start = start;
+    trunc_info->stop = stop;
+    trunc_info->orig_start_key = orig_start_key;
+    trunc_info->orig_stop_key = orig_stop_key;
+    if (uri != NULL)
+        trunc_info->uri = uri;
+    else if (start != NULL)
+        trunc_info->uri = start->internal_uri;
+    else {
+        WT_ASSERT(session, stop != NULL);
+        trunc_info->uri = stop->internal_uri;
+    }
```

Comment:
I oscillated about where the initialization block should be, and ended up with it lower down so it could be simpler.

I'm happy with this change, but I think it would be better again if the code here didn't overwrite the input `start` parameter - that seems excessively subtle.

Reply:
Do you mean further down at line 1621? I could add an assert that `trunc_info->start` is NULL there so that it is clear it isn't overwritten. Or you mean the code's original use of setting `start` when it is passed in as NULL?